libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -226,14 +226,16 @@ public static async Task<PromptRecognizerResult<TokenResponse>> RecognizeTokenAs
                     var serviceUrl = dc.Context.Activity.ServiceUrl;
                     var claimsIdentity = turnContext.TurnState.Get<ClaimsIdentity>(BotAdapter.BotIdentityKey);
                     var audience = callerInfo.Scope;
-                    var connectorClient = await UserTokenAccess.CreateConnectorClientAsync(turnContext, serviceUrl, claimsIdentity, audience, cancellationToken).ConfigureAwait(false);
-                    if (turnContext.TurnState.Get<IConnectorClient>() != null)
+                    var newConnectorClient = await UserTokenAccess.CreateConnectorClientAsync(turnContext, serviceUrl, claimsIdentity, audience, cancellationToken).ConfigureAwait(false);
+                    var oldConnectorClient = turnContext.TurnState.Get<IConnectorClient>();
+                    if (oldConnectorClient != null)

```

Comment:
```suggestion
                    var oldConnectorClient = turnContext.TurnState.Get<IConnectorClient>();
                    if (oldConnectorClient != null) {
                        oldConnectorClient.Dispose();
                    }
                    turnContext.TurnState.Set(newConnectorClient);
```

Small optimization that allows you to get rid of the `else` block and separate `Add()` call. But there's probably a more C# idiomatic way to do this.